luma/core/lib.py

```diff
@@ -20,8 +20,9 @@ def __rpi_gpio__(self):
     # a Raspberry Pi... this is imported here so we can swap out the
     # implementation for a mock
     try:
-        import RPi.GPIO
-        return RPi.GPIO
+        import RPi.GPIO as GPIO
+        GPIO.setmode(GPIO.BCM)
```

Comment:
For the default case, it does initialize RPi.GPIO with the BCM mode when `gpio=None` is supplied.

But when supplying a specific GPIO module to the spi class, we shouldn't enforce it to be BCM.... Say I have an existing application, which I've coded up to already use RPi.GPIO with BOARD mode, in order to make use of it I would need to pass in `gpio=RPi.GPIO`, and then supply `gpio_DC=<board pin>`, `gpio_RST=<board pin>`

Or in the case of orange Pi, I would initialize like:

```python
import OPi.GPIO as GPIO
from luma.core.serial import spi

GPIO.setmode(GPIO.SUNXI)

spi = serial(gpio=GPIO, gpio_DC="PA01", gpio_RST="PG06", ...)
```

Right now, the above code would not be able to work.

The `setmode` method has just moved, and would get initialized once per newly created `spi` class, or in the case of when `gpio=GPIO` is supplied it is at the end users discretion to set the mode.